### ISSUE TITLE
Keyboard focus does not remain on same control after pressing 'ESC' key to close the opened ‘Add page’ popup 

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#100]
User Experience:

If focus order is not sequential and meaningful, then keyboard user will find hard to understand the actual flow of page and the actual feature of the page which degrades the user experience.

Note: User credentials should NOT be included in the bug.

 

Repro Steps:

1. Open Azure portal with valid credentials and create a API Management.

2. Click Developer Portal button to navigate to the portal.

3. Wait portal to load in design mode.

4. Navigate to the Left nav and select Pages button.

5. Navigate to the "Pages" window and select "Add page" button.

6. Verify whether focus retain on ‘Add page’ button after pressing 'ESC' key for closing the opened ‘Add page’ screen.

 

Actual Result:

Keyboard focus does not remain on same control as ‘Add page’ button after pressing 'ESC' key to close the opened ‘Add page’ popup while navigating with keyboard. After pressing 'ESC' key whole popup as ‘pages and page’ gets closed.

 

Expected result:

Keyboard focus should remain on same control as ‘Add page’ button after closing the opened popup ‘Add page while navigating with keyboard. After pressing 'ESC' key only ‘Add page’ screen should close and focus should remain on ‘Add page’ button.

 

Note:

Similar issue exist with Chrome.

Similar issue exist throughout the application.

 

MAS Reference:

MAS 3.2.1 - On Focus (06)

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.
